### PR TITLE
Ruowan/add emum rule

### PR DESCRIPTION
**Pull request by ruowan**

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -1,8 +1,15 @@
 # Changelog
 
+## What's New (2/18/2020)
```
Comment (lirenhe) [inline changelog.md]: You may work with Jianye for the complete change log. I think we will only release one that combines the change from you and Jianye.

File: src/typescript/azure-openapi-validator/rules/EnumMustHaveType.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  id: "R3015",
+  name: EnumMustHaveType,
+  severity: "warning",
+  category: "ARMViolation",
+  mergeState: MergeStates.individual,
+  openapiType: OpenApiTypes.arm | OpenApiTypes.dataplane,
+  appliesTo_JsonQuery: "$..*[?(@.enum)]",
+  run: function*(doc, node, path) {
+    const msg: string = `Enum must define its type and "object" type is not allowed due to Autorest refuse to parse it.`;
+    if (node.type === undefined || node.type.toLowerCase() === "object") {
```
Comment (lirenhe) [inline src/typescript/azure-openapi-validator/rules/EnumMustHaveType.ts]: could you use !node.type instead?

File: src/typescript/azure-openapi-validator/rules/EnumUniqueValue.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+rules.push({
+  id: "R3024",
+  name: EnumUniqueValue,
+  severity: "warning",
+  category: "ARMViolation",
+  mergeState: MergeStates.individual,
+  openapiType: OpenApiTypes.arm | OpenApiTypes.dataplane,
+  appliesTo_JsonQuery: "$..*[?(@.enum)]",
+  run: function*(doc, node, path) {
+    const msg: string = `Enum must not contain case-insensitive duplicated value and make sure every value in enum unique.`;
```
Comment (lirenhe) [inline src/typescript/azure-openapi-validator/rules/EnumUniqueValue.ts]: could you update the message to contain enum value?

File: src/typescript/package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "@microsoft.azure/openapi-validator",
-    "version": "1.0.5",
+    "version": "1.2.0",
```
Comment (lirenhe) [inline src/typescript/package.json]: Please work with Jianye to change the version together and why use 1.2.0 directly?

File: changelog.md (hunk 1 above)
Comment (ruowan) [reply changelog.md]: Ok. I will talk about Jianye offline.

File: src/typescript/package.json (hunk 4 above)
Comment (ruowan) [reply src/typescript/package.json]: Ok. We will create a separate PR to bump version and update changelog.

File: src/typescript/azure-openapi-validator/rules/EnumMustHaveType.ts (hunk 2 above)
Comment (ruowan) [reply src/typescript/azure-openapi-validator/rules/EnumMustHaveType.ts]: I perfer ”node.type===undefined“ is more clear and means author forget to define its type. `!node.type` include `false, empty, null...`.

File: src/typescript/azure-openapi-validator/rules/EnumUniqueValue.ts (hunk 3 above)
Comment (ruowan) [reply src/typescript/azure-openapi-validator/rules/EnumUniqueValue.ts]: Updated.

File: src/typescript/azure-openapi-validator/rules/EnumMustHaveType.ts (hunk 2 above)
Comment (lirenhe) [reply src/typescript/azure-openapi-validator/rules/EnumMustHaveType.ts]: what is the type of node.type, is it string?
what if node.type is null?

Comment (ruowan) [reply src/typescript/azure-openapi-validator/rules/EnumMustHaveType.ts]: type is string.

**Pull request merged**